plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1805,6 +1818,26 @@ private ProtocolEntry getProtocolEntry(ConnectorSession session, SchemaTableName
         return metastore.getProtocol(session, metastore.getSnapshot(schemaTableName, session));
     }
 
+    private ProtocolEntry getProtocolEntry(Map<String, Object> properties)
+    {
+        Optional<Integer> readerVersion = getReaderVersion(properties);
+        Optional<Integer> writerVersion = getWriterVersion(properties);
+        Optional<Boolean> changeDataFeedEnabled = getChangeDataFeedEnabled(properties);
+
+        if (changeDataFeedEnabled.isPresent() && changeDataFeedEnabled.get()) {
+            writerVersion.ifPresent(version -> {
+                if (version < MAX_WRITER_VERSION) {
+                    throw new TrinoException(INVALID_TABLE_PROPERTY, WRITER_VERSION_PROPERTY + " cannot be set less than " + MAX_WRITER_VERSION + " when cdf is enabled");
+                }
+            });
+            // Enabling cdf (change data feed) requires setting the writer version to 4
+            writerVersion = Optional.of(CDF_SUPPORTED_WRITER_VERSION);
```

Comment:
I don't think we should do this. If the user sets the writer version to 2, when they do a `show create table` it should say 2.

Reply:
If the user sets the writer version to less than 4 and cdf is enabled then it will throw an exception.

Reply:
Got it, I see what you're doing now :+1:

Reply:
Though, if some has set writer version to 5 explicitly, this will set it to 4. I think you only want to do this if `writerVersion` hasn't been set explicitly

Reply:
> I think you only want to do this if writerVersion hasn't been set explicitly

Addressed.